src/resources/2015/Dinsey.ts

```diff
@@ -0,0 +1,221 @@
+  ],
+  ["questEStSocialJusticeI", $location`Pirates of the Garbage Barges`],
+  ["questEStFishTrash", $location`Pirates of the Garbage Barges`],
+  ["questEStZippityDooDah", $location`The Toxic Teacups`],
+  ["questEStSuperLuber", $location`Barf Mountain`],
+  ["questEStGiveMeFuel", $location`The Toxic Teacups`],
+  [
+    "questEStNastyBears",
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+  ["questEStWorkWithFood", $location`Barf Mountain`],
+]);
+
+interface QuestRequirement {
+  item: Item | null;
+  quantity: number;
+  trackingPref?: string;
+  completeValue?: number;
+}
+
+export const questRequirements = new Map<string, QuestRequirement>([
+  [
+    "questEStSocialJusticeII",
+    {
+      item: null,
+      quantity: 0,
+      trackingPref: "dinseySocialJusticeIIProgress",
+      completeValue: 15,
+    },
+  ],
+  [
+    "questEStSocialJusticeI",
+    {
+      item: null,
+      quantity: 0,
+      trackingPref: "dinseySocialJusticeIProgress",
+      completeValue: 15,
+    },
+  ],
+  [
+    "questEStFishTrash",
+    {
+      item: $item`trash net`,
+      quantity: 1,
+      trackingPref: "dinseyFilthLevel",
+      completeValue: 0,
+    },
+  ],
+  [
+    "questEStZippityDooDah",
+    {
+      item: $item`Dinsey mascot mask`,
+      quantity: 1,
+      trackingPref: "dinseyFunProgress",
+      completeValue: 15,
+    },
+  ],
+  [
+    "questEStSuperLuber",
+    {
+      item: $item`lube-shoes`,
+      quantity: 1,
+    },
+  ],
+  [
+    "questEStGiveMeFuel",
+    {
+      item: $item`toxic globule`,
+      quantity: 20,
+    },
+  ],
+  [
+    "questEStNastyBears",
+    {
+      item: null,
+      quantity: 0,
+      trackingPref: "dinseyNastyBearsDefeated",
+      completeValue: 8,
+    },
+  ],
+  [
+    "questEStWorkWithFood",
+    {
+      item: $item`complimentary Dinsey refreshments`,
+      quantity: 1,
+      trackingPref: "dinseyTouristsFed",
+      completeValue: 30,
+    },
+  ],
+]);
+
+export const keyCardsLocations = new Map<Item, Location>([
+  [$item`keycard α`, $location`Barf Mountain`],
+  [$item`keycard β`, $location`Pirates of the Garbage Barges`],
+  [$item`keycard γ`, $location`The Toxic Teacups`],
+  [
+    $item`keycard δ`,
+    $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`,
+  ],
+]);
+
+export function acceptedQuest(): string | null {
+  const result = Array.from(questPreferenceKioskText.values()).find(
+    (questPref) => getString(questPref) !== "unstarted"
+  );
+  if (result) {
+    return result;
+  } else {
+    return null;
+  }
+}
+
+export function questFinished() {
+  const quest = acceptedQuest();
+  return quest && getString(quest) === "finished";
+}
+
+export function availableQuests(): string[] {
+  const kioskText = visitUrl(kioskUrl);
+  return Array.from(questPreferenceKioskText)
+    .filter((questPair) => kioskText.includes(questPair[1]))
+    .map((questPair) => questPair[0]);
+}
+
+export function startQuest(questPref: string) {
```

Comment:
make this a StringProperty OR make a type consisting specifically of the Dinsey quest props